kytos/core/auth.py

```diff
@@ -136,232 +234,107 @@ def register_core_auth_services(self):
+        return answer
+
+    @staticmethod
+    def _get_request():
+        """Get JSON from request"""
+        try:
+            metadata = request.json
+            content_type = request.content_type
+        except BadRequest as err:
+            result = 'The request body is not a well-formed JSON.'
+            raise BadRequest(result) from err
+        if content_type is None:
+            result = 'The request body is empty.'
+            raise BadRequest(result)
+        if metadata is None:
+            if content_type != 'application/json':
+                result = ('The content type must be application/json '
+                          f'(received {content_type}).')
             else:
-                response = {
-                    "answer": {"users": boxes},
-                    "code": HTTPStatus.OK.value,
-                }
-
-        content = {
-            "namespace": self.namespace,
-            "callback": _list_users_callback,
-        }
-        event = KytosEvent(name="kytos.storehouse.list", content=content)
-        self.controller.buffers.app.put(event)
-        while True:
-            time.sleep(0.1)
-            if response:
-                break
-        return response["answer"], response["code"]
+                result = 'Metadata is empty.'
+            raise UnsupportedMediaType(result)
+        return metadata
 
     @authenticated
     def _create_user(self):
-        """Save a user using Storehouse."""
-        response = {}
-
-        def _create_user_callback(_event, box, error):
-            nonlocal response
-            if not box:
-                response = {
-                    "answer": 'User already exists',
-                    "code": HTTPStatus.CONFLICT.value,
-                }
-            elif error:
-                response = {
-                    "answer": "User has not been created",
-                    "code": HTTPStatus.INTERNAL_SERVER_ERROR.value,
-                }
-            else:
-                response = {
-                    "answer": "User successfully created",
-                    "code": HTTPStatus.OK.value,
-                }
-
-        req = request.json
-        password = req["password"].encode()
-        data = {
-            "username": req["username"],
-            "email": req["email"],
-            "password": hashlib.sha512(password).hexdigest(),
-        }
-        content = {
-            "namespace": self.namespace,
-            "box_id": data["username"],
-            "data": data,
-            "callback": _create_user_callback,
-        }
-        event = KytosEvent(name="kytos.storehouse.create", content=content)
-        self.controller.buffers.app.put(event)
-        while True:
-            time.sleep(0.1)
-            if response:
-                break
-        return response["answer"], response["code"]
+        """Save a user using MongoDB."""
+        user = self.user_controller.create_user(self._get_request())
```

Comment:
@Alopalao you need to catch the exceptions that `create_user` can raise and return/raise the status code accordingly. Same thing for `update_user`, thanks.

Reply:
For the errors, 409 (Conflict) when it's a duplicated user and 400 (BadRequest) otherwise.